### PR TITLE
fix(pipelineRef): add resolvedExpectedArtifacts from pipelineTrigger to PipelineRefTrigger

### PR DESCRIPTION
**Pull request by edgarulg**
Spel expressions with the function `#triggerResolvedArtifact` are throwing an evaluation error when pipelineRef feature is enabled in Orca. See: https://github.com/spinnaker/spinnaker/issues/6993

The error happens because when we transform a PipelineTrigger into a PipelineRefTrigger we are not populating the `resolvedExpectedArtifacts`. As the `resolvedExpectedArtifacts` is empty the Spel function `#triggerResolvedArtifact`  is not able to collect the artifact we want.

This small code change fix the issue.

### CODE REVIEW

**Comment by jasonmcintosh** [comment]
Tests please.

**Comment by edgarulg** [comment]
sure, let me add test around it

File: orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/PipelineRefTriggerDeserializerSupplier.kt [hunk 1]
```diff
@@ -61,7 +61,9 @@ class PipelineRefTriggerDeserializerSupplier(
               isStrategy = get("strategy")?.booleanValue() == true,
               parentExecutionId = parentExecutionId,
               parentPipelineStageId = get("parentPipelineStageId")?.textValue()
-            )
+            ).apply {
+              resolvedExpectedArtifacts = get("resolvedExpectedArtifacts")?.listValue(parser) ?: mutableListOf()
```
[the remark this answers is not in the document]
Comment (edgarulg) [reply orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/PipelineRefTriggerDeserializerSupplier.kt:65]: good point. The other field also should be added. I will add it as part of this PR as well.

**Comment by edgarulg** [comment]
@jasonmcintosh I added tests and PR is ready for review

**Review by jasonmcintosh — APPROVED** [review]
I think the tests are a TOUCH more complicated than they need to be but work well for waht this is.  Looks like this should fix dealing with the resolved expected artifacts...

**Comment by jasonmcintosh** [comment]
NOT sure we won't hit more cases but... kinda hoping we can keep this as small as possible...

**Comment by jasonmcintosh** [comment]
@mergifyio backport release-1.36.x release-1.35.x

**Pull request merged**